playwright.config.ts

```diff
@@ -0,0 +1,32 @@
+import { devices, PlaywrightTestConfig } from '@playwright/test';
+
+// Config file docs: https://playwright.dev/docs/test-configuration
+const config: PlaywrightTestConfig = {
+  webServer: {
+    command: 'pnpm start',
+    port: 3000,
+    reuseExistingServer: !process.env.CI,
+  },
+  testIgnore: '**/util/__tests__/**',
+  retries: process.env.CI ? 2 : 0,
+  reporter: process.env.CI
+    ? 'list'
+    : [['html', { outputFolder: 'playwright/report/' }]],
```

Comment:
This defines the reporter to be used for test result output. In CI, we use the [list reporter](https://playwright.dev/docs/test-reporters#list-reporter). Otherwise, we use the [html reporter](https://playwright.dev/docs/test-reporters#html-reporter) and specify the output folder as `playwright/report/`.
https://playwright.dev/docs/api/class-testconfig#test-config-reporter

Reply:
- can you add the benefit of configuring `html` for local and `list` for CI to the comment above?
- this config should be the same config we use in student projects

Reply:
The benefits of the different reporters are:

- Locally: The `html` reporter is the best option because it collects all the data when a test fails, including the logs, the exact failure position, a screenshot, and a video. This information can help debug the test failure.
- CI: The `list` reporter is the best option for CI because it provides a clear overview of the test results. The failing tests are marked with an `X`, making it easy to see which failed. The `dots` reporter can also be used, but it does not provide as much information as the list reporter.

### list
```
 1  ✓ should access error in env (438ms)
 2  ✓ handle long test names (515ms)
 3  x 1) render expected (691ms)
 4  ✓ should timeout (932ms)
 5    should repeat each:
 6  ✓ should respect enclosing .gitignore (569ms)
 7    should teardown env after timeout:
 8    should respect excluded tests:
 9  ✓ should handle env beforeEach error (638ms)
```
 ### dots
```
······F·············································
```

Reply:
nice, thanks for the update! `'github'` looks good 👍